packages/gestalt/src/BannerOverlay.js

```diff
@@ -0,0 +1,276 @@
+// @flow strict
+import { Children, type Element, type ElementConfig, type Node as ReactNode } from 'react';
+import Avatar from './Avatar';
+import styles from './BannerOverlay.css';
+import PrimaryAction from './BannerOverlay/PrimaryAction';
+import {
+  BannerOverlayAvatarThumbnail,
+  BannerOverlayIconThumbnail,
+  BannerOverlayImageThumbnail,
+  BannerOverlayMessage,
+} from './BannerOverlay/subcomponents';
+import Box from './Box';
+import Button from './Button';
+import ButtonLink from './ButtonLink';
+import { useColorScheme } from './contexts/ColorSchemeProvider';
+import { useDefaultLabelContext } from './contexts/DefaultLabelProvider';
+import { useDeviceType } from './contexts/DeviceTypeProvider';
+import Flex from './Flex';
+import Icon from './Icon';
+import IconButton from './IconButton';
+import Image from './Image';
+import Layer from './Layer';
+import Link from './Link';
+import Text from './Text';
+import { type Indexable } from './zIndex';
+
+const DEFAULT_COLORS = {
+  lightModeBackground: 'default',
+  darkModeBackground: 'elevationRaised',
+  textColor: 'default',
+  iconColor: 'white',
+};
+
+type Props = {
+  /**
+   * Adds a dismiss button to BannerOverlay. See the [Dismissible variant](https://gestalt.pinterest.systems/web/banneroverlay#Dismissible) for more info.
+   * The `accessibilityLabel` should follow the [Accessibility guidelines](https://gestalt.pinterest.systems/web/banneroverlay#Accessibility).
+   *
+   */
+  dismissButton?: {
+    accessibilityLabel?: string,
+    onDismiss: () => void,
+  },
+  /**
+   * Main content of BannerOverlay. Content should be [localized](https://gestalt.pinterest.systems/web/banneroverlay#Localization). See the [Text variant](https://gestalt.pinterest.systems/web/banneroverlay#Text) to learn more.
+   */
+  message: string | Element<typeof Text>,
+  /**
+   * Distance (in pixels) from the viewport edge (offsetTop will be used, if desktop, offsetBottom will be used, if mobile).
+   */
+  offsetTop?: number,
+  /**
+   * Distance (in pixels) from the viewport edge (offsetTop will be used, if desktop, offsetBottom will be used, if mobile).
+   */
+  offsetBottom?: number,
+  /**
+   * Helper [Link](https://gestalt.pinterest.systems/web/link) to be placed after the subtext. See the [helper link variant](https://gestalt.pinterest.systems/web/banneroverlay#helperLink) to learn more.
+   */
+  helperLink?: {
+    text: string,
+    accessibilityLabel: string,
+    href: string,
+    onClick?: ({
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    }) => void,
+  },
+  /**
+   * Adds an optional primary button for user interaction.
```

Comment:
See `OnLinkNavigationProvider`... add link here

Reply:
This probably refers to some old handler for Link that got pasted here during documentation. I'll double check with Cintia, but I believe this can just be removed.